include/swift/AST/DiagnosticsSema.def

```diff
@@ -4430,6 +4430,9 @@ WARNING(redundant_particular_literal_case,none,
 NOTE(redundant_particular_literal_case_here,none,
      "first occurrence of identical literal pattern is here", ())
 
+ERROR(case_as_punct,none,
+      "'as' in 'case' statement dose not use '%select{?|!}0'", (bool))
```

Comment:
Typo: "does"

@xedin, what do you think about this diagnostic phrasing? This seems okay, but I have a feeling like it could be a little more helpful too, though I can't put my finger on it.

Reply:
Maybe something like `'as' in 'case' statement always implies conditional downcast'?

Reply:
It's the other way around, though. They wrote `as?` when they should have written `as`.

Reply:
I understand, that's what I tried to diagnose :) How about - `use cast operator 'as' to represent a conditional downcast in 'case' statement` for `as?`, and `cannot use 'as!' in 'case' statement because it only supports conditional downcast'